src/App.tsx

```diff
@@ -2,50 +1,107 @@
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
+interface Good {
+  id: number;
+  name: string;
+}
+
+enum SortBy {
+  abc = 'abc',
+  length = 'length',
+}
+
+function getReorderedGoods(
+  goods: Good[],
+  sortType: string,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortBy.length) {
+    visibleGoods.sort((good1, good2) => good1.name.length - good2.name.length);
+  }
+
+  if (sortType === SortBy.abc) {
+    visibleGoods.sort((good1, good2) => good1.name.localeCompare(good2.name));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortField, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortBy.abc,
+          })}
+          onClick={() => setSortField(SortBy.abc)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortBy.length,
+          })}
+          onClick={() => setSortField(SortBy.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': reversed !== true,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField || reversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        ) : null}
       </div>
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
+      <GoodsList goods={visibleGoods} />
```

Comment:
Ensure that the `GoodsList` component is correctly receiving the `goods` prop with the expected type. This line seems correct, but verify that the `GoodsList` component is implemented to handle the `goods` prop properly.

Reply:
please give an example